lxc/storage.go

```diff
@@ -0,0 +1,861 @@
+		fmt.Printf(i18n.G("Storage pool %s deleted")+"\n", name)
+	}
+
+	return err
+}
+
+func (c *storageCmd) doStoragePoolEdit(client *lxd.Client, name string) error {
+	// If stdin isn't a terminal, read text from it
+	if !termios.IsTerminal(int(syscall.Stdin)) {
+		contents, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return err
+		}
+
+		newdata := api.StoragePool{}
+		err = yaml.Unmarshal(contents, &newdata)
+		if err != nil {
+			return err
+		}
+		return client.StoragePoolPut(name, newdata)
+	}
+
+	// Extract the current value
+	pool, err := client.StoragePoolGet(name)
+	if err != nil {
+		return err
+	}
+
+	data, err := yaml.Marshal(&pool)
+	if err != nil {
+		return err
+	}
+
+	// Spawn the editor
+	content, err := shared.TextEditor("", []byte(c.storagePoolEditHelp()+"\n\n"+string(data)))
+	if err != nil {
+		return err
+	}
+
+	for {
+		// Parse the text received from the editor
+		newdata := api.StoragePool{}
+		err = yaml.Unmarshal(content, &newdata)
+		if err == nil {
+			err = client.StoragePoolPut(name, newdata)
+		}
+
+		// Respawn the editor
+		if err != nil {
+			fmt.Fprintf(os.Stderr, i18n.G("Config parsing error: %s")+"\n", err)
+			fmt.Println(i18n.G("Press enter to open the editor again"))
+
+			_, err := os.Stdin.Read(make([]byte, 1))
+			if err != nil {
+				return err
+			}
+
+			content, err = shared.TextEditor("", content)
+			if err != nil {
+				return err
+			}
+			continue
+		}
+		break
+	}
+	return nil
+}
+
+func (c *storageCmd) doStoragePoolGet(client *lxd.Client, name string, args []string) error {
+	// we shifted @args so so it should read "<key>"
+	if len(args) != 1 {
+		return errArgs
+	}
+
+	resp, err := client.StoragePoolGet(name)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range resp.PoolConfig {
+		if k == args[0] {
+			fmt.Printf("%s\n", v)
+		}
+	}
+	return nil
+}
+
+func (c *storageCmd) doStoragePoolsList(config *lxd.Config, args []string) error {
+	var remote string
+	if len(args) > 1 {
+		var name string
+		remote, name = config.ParseRemoteAndContainer(args[1])
+		if name != "" {
+			return fmt.Errorf(i18n.G("Cannot provide container name to list"))
+		}
+	} else {
+		remote = config.DefaultRemote
+	}
+
+	client, err := lxd.NewClient(config, remote)
+	if err != nil {
+		return err
+	}
+
+	pools, err := client.ListStoragePools()
+	if err != nil {
+		return err
+	}
+
+	data := [][]string{}
+	for _, pool := range pools {
+		size := pool.PoolConfig["size"]
+		sz, err := strconv.ParseUint(pool.PoolConfig["size"], 10, 64)
+		if err == nil {
+			size = shared.GetByteSizeString(int64(sz), 0)
+		}
+
+		usedby := strconv.Itoa(len(pool.PoolUsedBy))
+
+		data = append(data, []string{pool.PoolName, pool.PoolDriver, size, pool.PoolConfig["source"], usedby})
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetAutoWrapText(false)
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetRowLine(true)
+	table.SetHeader([]string{
+		i18n.G("NAME"),
+		i18n.G("DRIVER"),
+		i18n.G("SIZE"),
```

Comment:
Do we actually have something useful to report for SIZE? I thought we said we'd drop it from the list since the pool size is outside our control and difficult to figure out with some backends.